### PR TITLE
ISAICP-5951: Subscription report

### PR DESCRIPTION
**Pull request by pfrenssen**

### CODE REVIEW

File: web/modules/custom/joinup_subscription/joinup_subscription.routing.yml [hunk 1]
```diff
@@ -12,14 +53,6 @@ joinup_subscription.subscriptions:
...
         type: entity:user
 
-joinup_subscription.my_subscriptions:
-  path: '/user/subscriptions'
-  defaults:
-    _form: Drupal\joinup_subscription\Form\SubscriptionsForm
-    _title: 'My subscriptions'
-  requirements:
-    _user_is_logged_in: 'TRUE'
-
```
Comment (pfrenssen) [inline web/modules/custom/joinup_subscription/joinup_subscription.routing.yml:22]: This was moved up to maintain alphabetical ordering.

File: web/profiles/joinup/src/Form/SolutionsByLicenceForm.php [hunk 2]
```diff
@@ -99,8 +99,15 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
-    $total = reset($results)->total->getValue();
+
+    // Under some unknown circumstances the result returned can be a string with
+    // the value 'binding' instead of the expected count. This has been causing
+    // random test failures, and has also been observed during debugging on a
+    // development environment. It appears that calling ::getArrayCopy() works
+    // around this somehow and the problem can no longer be replicated.
+    $results = $this->connection->query($this->getCountQuery($licence_id))->getArrayCopy();
+    $result = reset($results);
+    $total = $result->total->getValue();
```
Comment (pfrenssen) [inline web/profiles/joinup/src/Form/SolutionsByLicenceForm.php:110]: This is an unrelated bugfix. This caused a random failure during development of this PR.

File: composer.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
         "php": ">=7.1.0",
         "ext-dom": "*",
         "ext-json": "*",
+        "ext-pdo": "*",
```
Comment (pfrenssen) [inline composer.json:10]: We're already using PDO but this was not declared in the Composer manifest. PHPStorm was complaining about this.

**Review by idimopoulos — CHANGES_REQUESTED** [review]

File: web/modules/custom/joinup_group/src/Entity/GroupTrait.php [hunk 4]
```diff
@@ -17,12 +19,27 @@ trait GroupTrait {
...
+    $uid = $uid ?? \Drupal::currentUser()->id();
+
     return $this->getMembershipManager()->getMembership($this, $uid, $states);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getGroupAccess(string $permission, ?AccountInterface $user = NULL): AccessResultInterface {
+    assert(is_subclass_of($this, ContentEntityBase::class), __TRAIT__ . ' is intended to be used in bundle classes for content entities.');
```
[the remark this answers is not in the document]
Comment (pfrenssen) [reply web/modules/custom/joinup_group/src/Entity/GroupTrait.php:35]: Good catch! This is probably originated from copy / pasting some other method and I forgot to update the message.

**Pull request merged**